examples/hybrid/driver.jl

```diff
@@ -156,8 +157,11 @@ function additional_cache(Y, params, model_spec, dt; use_tempest_mode = false)
             diffuse_momentum,
             coupled,
         ) : NamedTuple(),
-        model_spec.non_orographic_gravity_wave ? gravity_wave_cache(Y, FT) :
-        NamedTuple(),
+        model_spec.non_orographic_gravity_wave ?
+        (
+            config == "column" ? gravity_wave_column_cache(Y, FT) :
+            gravity_wave_sphere_cache(Y, FT)
+        ) : NamedTuple(),
```

Comment:
Once the type above is defined, we could dispatch on `gravity_wave_cache(config::ColumnConfig)`/`gravity_wave_cache(config::SphereConfig)` etc.

Or if these parameterizations are fundamentally different, we can add a type for the particular GWP and dispatch on them